scrapy_poet/commands.py

```diff
@@ -53,34 +52,44 @@ def __init__(self, crawler: Crawler) -> None:
         )
 
 
-def spider_for(injectable: Type[ItemPage]) -> Type[scrapy.Spider]:
-    class InjectableSpider(scrapy.Spider):
+def spider_for(
+    injectable: Type[ItemPage],
+    url: str,
+    base_spider: Optional[Type[scrapy.Spider]] = None,
+) -> Type[scrapy.Spider]:
+    if base_spider is None:
+        base_spider = scrapy.Spider
+
+    class InjectableSpider(base_spider):  # type: ignore[valid-type, misc]
         name = "injectable"
-        url = None
 
-        def start_requests(self):
-            yield scrapy.Request(self.url, self.cb)
+        def __init__(self, name=None, **kwargs):
+            super().__init__(name, **kwargs)
+            meta = {"savefixture": True}
+            self.start_requests = lambda: [scrapy.Request(url, self.cb, meta=meta)]
 
         async def cb(self, response: DummyResponse, page: injectable):  # type: ignore[valid-type]
             global frozen_time
             frozen_time = datetime.datetime.now(datetime.timezone.utc).replace(
                 microsecond=0
             )
             with time_machine.travel(frozen_time):
-                yield await ensure_awaitable(page.to_item())  # type: ignore[attr-defined]
+                item = await ensure_awaitable(page.to_item())  # type: ignore[attr-defined]
+            saved_items.append(item)
```

Comment:
Would it be worth it bringing back `SavingPipeline` to make it less fragile on cases like a custom middleware is enabled in the custom settings which replaces the callback method that was originally passed in `start_requests()`?

Reply:
Aha, you're right. 👍